File: backend/api/schemas/sticker.py
"""
贴纸生成 API 的 Pydantic 模型定义
"""
from pydantic import BaseModel, Field
from typing import Optional
from enum import Enum


class AIModelType(str, Enum):
    """AI 抠图模型类型"""
    BIREFNET = "birefnet"
    REMBG = "rembg"


class StickerGenerateRequest(BaseModel):
    """贴纸生成请求参数"""
    file_id: str = Field(..., description="上传文件的临时 ID")
    outline_width: int = Field(default=15, ge=5, le=50, description="白边宽度 (5-50 像素)")
    model_type: AIModelType = Field(default=AIModelType.BIREFNET, description="AI 抠图模型类型")


class UploadResponse(BaseModel):
    """文件上传响应"""
    file_id: str = Field(..., description="临时文件 ID")
    filename: str = Field(..., description="原始文件名")
    preview_url: str = Field(..., description="原图预览 URL")


class GenerateResponse(BaseModel):
    """贴纸生成响应"""
    task_id: str = Field(..., description="生成任务 ID")
    status: str = Field(..., description="任务状态: processing, completed, failed")
    preview_url: Optional[str] = Field(None, description="预览图 URL")
    pdf_url: Optional[str] = Field(None, description="PDF 下载 URL")
    svg_url: Optional[str] = Field(None, description="SVG 下载 URL")
    error: Optional[str] = Field(None, description="错误信息")


class TaskStatusResponse(BaseModel):
    """任务状态响应"""
    task_id: str
    status: str
    progress: int = Field(default=0, ge=0, le=100, description="进度百分比")
    preview_url: Optional[str] = None
    pdf_url: Optional[str] = None
    svg_url: Optional[str] = None
    error: Optional[str] = None

File: frontend/src/api/sticker.js
/**
 * 贴纸 API 调用封装
 */
import axios from 'axios'

// API 基础地址 (使用 Vite 代理时为空，生产环境可配置)
const API_BASE = ''

const api = axios.create({
  baseURL: API_BASE,
  timeout: 60000, // 60秒超时 (模型处理可能较慢)
})

/**
 * 上传图片
 * @param {File} file - 图片文件
 * @returns {Promise<{file_id: string, filename: string, preview_url: string}>}
 */
export async function uploadImage(file) {
  const formData = new FormData()
  formData.append('file', file)

  const response = await api.post('/api/sticker/upload', formData, {
    headers: {
      'Content-Type': 'multipart/form-data',
    },
  })
  return response.data
}

/**
 * 生成贴纸
 * @param {Object} params - 生成参数
 * @param {string} params.file_id - 上传文件 ID
 * @param {number} params.outline_width - 白边宽度
 * @param {string} params.model_type - AI 模型类型 (birefnet / rembg)
 * @returns {Promise<{task_id: string, status: string}>}
 */
export async function generateSticker(params) {
  const response = await api.post('/api/sticker/generate', params)
  return response.data
}

/**
 * 获取任务状态
 * @param {string} taskId - 任务 ID
 * @returns {Promise<{task_id: string, status: string, progress: number, preview_url: string, pdf_url: string, svg_url: string}>}
 */
export async function getTaskStatus(taskId) {
  const response = await api.get(`/api/sticker/status/${taskId}`)
  return response.data
}

/**
 * 获取下载链接
 * @param {string} taskId - 任务 ID
 * @param {string} fileType - 文件类型 (preview / pdf / svg)
 * @returns {string} 下载 URL
 */
export function getDownloadUrl(taskId, fileType) {
  return `${API_BASE}/api/sticker/download/${taskId}/${fileType}`
}

/**
 * 获取原图预览链接
 * @param {string} fileId - 文件 ID
 * @returns {string} 预览 URL
 */
export function getOriginalPreviewUrl(fileId) {
  return `${API_BASE}/api/sticker/preview/original/${fileId}`
}

export default api
